acceptor/tcp_acceptor.go

```diff
@@ -78,21 +77,26 @@ func (t *tcpPlayerConn) GetNextMessage() (b []byte, err error) {
 
 // NewTCPAcceptor creates a new instance of tcp acceptor
 func NewTCPAcceptor(addr string, certs ...string) *TCPAcceptor {
-	keyFile := ""
-	certFile := ""
+	certificates := []tls.Certificate{}
 	if len(certs) != 2 && len(certs) != 0 {
 		panic(constants.ErrInvalidCertificates)
 	} else if len(certs) == 2 {
-		certFile = certs[0]
-		keyFile = certs[1]
+		cert, err := tls.LoadX509KeyPair(certs[0], certs[1])
+		if err != nil {
+			panic(constants.ErrInvalidCertificates)
+		}
+		certificates = append(certificates, cert)
 	}
 
+	return NewTCP(addr, certificates...)
+}
+
+func NewTCP(addr string, certs ...tls.Certificate) *TCPAcceptor {
```

Comment:
Given that `ListenAndServeTLS` requires providing a certificate and key, if you want to use this constructor, you would need to provide the same certificate information here and in the listen method? In more general terms, I cannot configure all certificates here and then just call the listen method without duplicating some of the certificates information, right?

Reply:
This behavior seems to be the same before the changes as well, to be honest.

Reply:
I don't know exactly why we have this one, maybe we should add a similar method that receives the certs. Anyway, the `ListAndServe` method already check for the certificates given through the `New...` methods.

Reply:
Oh, I see, the `ListenAndServe` will fall back to the TLS variant if there are any certificates provided in the constructor. LGTM, then.